django_auth_adfs/backend.py

```diff
@@ -15,19 +15,22 @@
 
 
 class AdfsBaseBackend(ModelBackend):
-    def exchange_auth_code(self, authorization_code, request):
-        logger.debug("Received authorization code: %s", authorization_code)
-        data = {
-            'grant_type': 'authorization_code',
-            'client_id': settings.CLIENT_ID,
-            'redirect_uri': provider_config.redirect_uri(request),
-            'code': authorization_code,
-        }
-        if settings.CLIENT_SECRET:
-            data['client_secret'] = settings.CLIENT_SECRET
 
-        logger.debug("Getting access token at: %s", provider_config.token_endpoint)
-        response = provider_config.session.post(provider_config.token_endpoint, data, timeout=settings.TIMEOUT)
+    def _ms_request(self, action, url, data=None, **kwargs):
```

Comment:
I tried running this code, there no handling for params and headers in request

Reply:
Can you share your code in a gist or a link to your repo with the change?

Reply:
`    def _ms_request(self, action, url, data=None, params=None,headers=None, **kwargs):
        """
        Make a Microsoft Entra/GraphQL request


        Args:
            action (callable): The callable for making a request.
            url (str): The URL the request should be sent to.
            data (dict): Optional dictionary of data to be sent in the request.

        Returns:
            response: The response from the server. If it's not a 200, a
                      PermissionDenied is raised.
        """
        response = action(url,params=params, headers=headers,data=data, timeout=settings.TIMEOUT, **kwargs)
        # 200 = valid token received
        # 400 = 'something' is wrong in our request
        if response.status_code == 400:
            if response.json().get("error_description", "").startswith("AADSTS50076"):
                raise MFARequired
            logger.error("ADFS server returned an error: %s", response.json()["error_description"])
            raise PermissionDenied

        if response.status_code != 200:
            logger.error("Unexpected ADFS response: %s", response.content.decode())
            raise PermissionDenied
        return response`

Reply:
Can you share your code in a gist or a link to your repo with the change?

Reply:
Also then calling from get_group_memberships_from_ms_graph change to

headers["ConsistencyLevel"] = "eventual"
response = self._ms_request(
    action=provider_config.session.get,
    url=graph_url,
    data=None,
    params=self.get_group_memberships_from_ms_graph_params(),
    headers=headers,
)

Reply:
https://github.com/vidhuUC/django-auth-adfs/commit/d46ebbe45141cee16a59d683c2ace67902fcfeee

Reply:
The doesn't appear to be using this change.